orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy

```diff
@@ -229,8 +230,45 @@ class FindImageFromClusterTask extends AbstractCloudProviderAwareTask implements
       }
     }.flatten()
 
+    List<Artifact> artifacts = imageSummaries.collect { placement, summaries ->
+      Artifact artifact = new Artifact()
+      summaries.findResults { summary ->
+        if (config.imageNamePattern && !(summary.imageName ==~ config.imageNamePattern)) {
+          return null
+        }
+        def location = "global"
+
+        if (placement.type == Location.Type.REGION) {
+          location = placement.value
+        } else if (placement.type == Location.Type.ZONE) {
+          location = placement.value
+        }
+
+        def metadata = [
+          sourceServerGroup: summary.serverGroupName,
+          refId: stage.refId
+        ]
+
+        try {
+          metadata.putAll(summary.image ?: [:])
+          metadata.putAll(summary.buildInfo ?: [:])
+        } catch (Exception e) {
+          log.error("Unable to merge server group image/build info (summary: ${summary})", e)
+        }
+
+        artifact.metadata = metadata
+        artifact.name = summary.imageName
+        artifact.location = location
+        artifact.type = "${cloudProvider}/image"
+        artifact.reference = "${summary.imageId}"
+        artifact.uuid = UUID.randomUUID().toString()
+      }
+      return artifact
```

Comment:
This seems a little odd - instead of assigning to `artifact` created in the parent scope, I would create artifact here, and then return that. Then assign the result of `findResults {..}.flatten` to something like `List<Artifact> artifacts`